### PR TITLE
converted donut-chart directive to component

### PR DESCRIPTION
**Pull request by kamleshverma1**
- converted directive to component for donut-chart

Signed-off-by: kamleshKumarVerma <kaverma@redhat.com>

### CODE REVIEW

**Comment by kamleshverma1** [comment]
@gnehapk @cloudbehl Please review it

File: src/modules/cluster/cluster-list/cluster-list.html [hunk 1]
```diff
@@ -64,8 +64,7 @@ <h1 class="bold-text">Clusters</h1>
             </div>
             <div class="col-md-1 text-center cluster-name bold-text counts" ng-click="clusterCntrl.goToClusterDetail(cluster.id)">{{cluster.name}}</div>
             <div class="col-md-3">
```
Comment (cloudbehl) [inline src/modules/cluster/cluster-list/cluster-list.html]: @kamleshverma1 as we discuss in the morning call. Please pass the id's or classes to the component for behavior test.

File: src/modules/host/host-list/host-list.html [hunk 2]
```diff
@@ -82,16 +82,13 @@ <h5 ng-if="hostCntrl.hostList.length !== 0">{{hostCntrl.filteredHostList.length}
...
                 </div>
                 <div  class="col-md-2">
-                    <div class="list-view-pf-additional-info-item donut-chart" chart-title="CPU" chart-data="host.cpu">
-                    </div>
+                    <donut-chart class="list-view-pf-additional-info-item" chart-title="CPU" chart-data="host.cpu"></donut-chart>
                 </div>
                 <div  class="col-md-2">
-                    <div class="list-view-pf-additional-info-item donut-chart" chart-title="Memory" chart-data="host.memory">
-                    </div>
+                    <donut-chart class="list-view-pf-additional-info-item" chart-title="Memory" chart-data="host.memory"></donut-chart>
```
Comment (cloudbehl) [inline src/modules/host/host-list/host-list.html]: Same as above. Pass the id's and classes.

File: src/commons/components/donut-chart/donut-chart.js [hunk 3]
```diff
@@ -42,69 +42,62 @@
...
+                controller: function($element, config) {
+                    var vm = this;
+                    vm.$postLink = function() {
+                        if(typeof vm.chartData !== "undefined") {
+                            var c3ChartDefaults = $().c3ChartDefaults();
+                            vm.customDonutChartConfig = c3ChartDefaults.getDefaultDonutConfig(vm.chartData["percent_used"]+"%");
+                            vm.customDonutChartConfig.size.height = 90;
+                            vm.customDonutChartConfig.size.width = 80;
+                            vm.customDonutChartConfig.donut.width = 10;
+                            vm.customDonutChartConfig.bindto = $element[0].childNodes[0]
```
Comment (cloudbehl) [inline src/commons/components/donut-chart/donut-chart.js]: Missed semi-colon

File: src/commons/components/donut-chart/donut-chart.js [hunk 4]
```diff
@@ -42,69 +42,62 @@
...
-                            order: null
-                        };
-                        c3.generate(scope.customDonutChartConfig);
-                    }
-
-                },
-                controller: function($scope, config) {
+                controllerAs: 'vm',
+                
+                controller: function($element, config) {
```
Comment (cloudbehl) [inline src/commons/components/donut-chart/donut-chart.js]: @kamleshverma1 I will recommend passing a name of controller here and writing the function differently.
Eg: 
controller: donutChartController,

function donutChartController($element, config){}

File: src/modules/pool/pool-list/pool-list.html [hunk 5]
```diff
@@ -67,7 +67,9 @@ <h1 class="bold-text">Pools</h1>
             </div>
             <div class="col-md-1 text-center bold-text pool-name">{{pool.name}}</div>
             <div class="col-md-3">
-                {{pool.utilization}}
+                
+                <donut-chart class="list-view-pf-additional-info-item" chart-title="Storage" chart-data="pool.utilization"></donut-chart>
```
Comment (cloudbehl) [inline src/modules/pool/pool-list/pool-list.html]: Same as above. Pass the id's and classes.

**Comment by kamleshverma1** [comment]
@cloudbehl re-worked on your comments .. please have look again

File: src/commons/components/donut-chart/donut-chart.js [hunk 6]
```diff
@@ -42,69 +42,62 @@
...
-                            order: null
-                        };
-                        c3.generate(scope.customDonutChartConfig);
-                    }
-
-                },
-                controller: function($scope, config) {
+                controllerAs: 'vm',
+                
+                controller: function donutChartController($element, config) {
```
Comment (cloudbehl) [inline src/commons/components/donut-chart/donut-chart.js]: just write name of the controller.

Controller: donutChartController,

donutChartController = function ($element, config){}

**Pull request merged**